### PR TITLE
onboard Forcepoint Cloud Security Gateway

### PR DESCRIPTION
**Pull request by Dlo-Bagari**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: DataConnectors/Forcepoint Cloud Security Gateway.json [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    "publisher": "Forcepoint",
+    "descriptionMarkdown": "Forcepoint Cloud Security Gateway is a converged cloud security service that provides visibility, control, and threat protection for users and data, wherever they are. For more information visit: https://www.forcepoint.com/product/cloud-security-gateway",
+    "graphQueries": [
+        {
+            "metricName": "Total data received",
+            "legend": "Forcepoint CSG",
+            "baseQuery": "\nCommonSecurityLog\n| where DeviceVendor == \"Forcepoint CSG\"\n"
+        }
+    ],
+    "sampleQueries": [
```
Comment (v-maudan) [inline DataConnectors/Forcepoint Cloud Security Gateway.json:13]: Remove one extra space from all sample queries after TimeGenerated.

File: DataConnectors/Forcepoint Cloud Security Gateway.json [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            "title": "Validate connection",
+            "description": "Follow the instructions to validate your connectivity:\n\nOpen Log Analytics to check if the logs are received using the CommonSecurityLog schema.\n\n>It may take about 20 minutes until the connection streams data to your workspace.\n\nIf the logs are not received, run the following connectivity validation script:\n\n> 1. Make sure that you have Python on your machine using the following command: python -version \n \n>2. You must have elevated permissions (sudo) on your machine",
+            "instructions": [
+                {
+                    "parameters": {
+                        "fillWith": [
+                            "WorkspaceId"
+                        ],
+                        "label": "Run the following command to validate your connectivity:",
+                        "value": "sudo wget https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/CEF/cef_troubleshoot.py&&sudo python cef_troubleshoot.py  {0}"
```
Comment (v-maudan) [inline DataConnectors/Forcepoint Cloud Security Gateway.json:107]: -O is missing in the command. Please refer template https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_CEF_template.json

File: DataConnectors/Forcepoint Cloud Security Gateway.json [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    "write": true,
+                    "delete": true
+                }
+            }
+        ]
+    },
+    "instructionSteps": [
+        {
+            "title": "Linux Syslog agent configuration",
+            "description": "This integration requires the Linux Syslog agent to collect your Forcepoint Cloud Security Gateway Web/Email logs as Common Event Format (CEF) and forward them to Azure Sentinel.",
```
Comment (v-maudan) [inline DataConnectors/Forcepoint Cloud Security Gateway.json:68]: CEF connector should communicate via some port (514 TCP default).Please refer template https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_CEF_template.json

File: Workbooks/Forcepoint Cloud Secuirty Gateway workbook.json [hunk 4]
```diff
@@ -0,0 +1,158 @@
+{
```
Comment (v-maudan) [inline Workbooks/Forcepoint Cloud Secuirty Gateway workbook.json:1]: workbooksMetadata.json file should have an entry for this workbook. for update please refer this link. https://github.com/Azure/Azure-Sentinel/tree/master/Workbooks

**Review by v-maudan — CHANGES_REQUESTED** [review]

File: Workbooks/Forcepoint Cloud Secuirty Gateway workbook.json [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      "content": {
+        "json": "## Top 5 Web requested Domains with log severity equal to 6 (Medium)\n---"
+      },
+      "name": "text - 2"
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "CommonSecurityLog\n| where TimeGenerated  <= ago(0m)\n| where DeviceVendor == \"Forcepoint CSG\"\n| where DeviceProduct == \"Web\"\n| where LogSeverity == 6\n| where DeviceCustomString2 != \"\"\n| summarize Count=count() by DeviceCustomString2\n| top 5 by Count\n| render piechart",
```
Comment (v-maudan) [inline Workbooks/Forcepoint Cloud Secuirty Gateway workbook.json:15]: There should be at least 4 charts / graphs in the workbook.(Currently there are 3 charts/graphs present)

File: DataConnectors/Forcepoint Cloud Security Gateway.json [hunk 6]
```diff
@@ -0,0 +1,118 @@
+{
```
Comment (v-maudan) [inline DataConnectors/Forcepoint Cloud Security Gateway.json:1]: Data connector json file should be in the format ProviderNameApplianceName.json . Please remove the space from DataConnector file name.

File: Workbooks/Forcepoint Cloud Secuirty Gateway workbook.json (hunk 4 above)
Comment (v-maudan) [inline Workbooks/Forcepoint Cloud Secuirty Gateway workbook.json:1]: No space in file name please.
Rename the workbook file name to ForcepointCloudSecuirtyGatewayworkbook.json

**Comment by v-jayakal** [comment]
@Dlo-Bagari, Can you please update the PR on comment resolutions? Thanks.

File: Workbooks/Forcepoint Cloud Secuirty Gateway workbook.json (hunk 4 above)
Comment (v-maudan) [reply Workbooks/Forcepoint Cloud Secuirty Gateway workbook.json]: Need to update the Workbook metadata file WorkbooksMetadata.json to include the new workbook. Please add workbook metadata definition.

Comment (v-maudan) [reply Workbooks/Forcepoint Cloud Secuirty Gateway workbook.json]: @Dlo-Bagari , Can you please update on this.

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: Workbooks/WorkbooksMetadata.json [hunk 7]
```diff
@@ -1088,5 +1088,18 @@
...
     "subtitle": "",
     "provider": "Intsights Cyber Intelligence"
-  }
+  },
+  {
+ "workbookKey": "ForcepointCloudSecurityGatewayWorkbook",
+ "logoFileName": "Forcepoint_new_logo.svg",
+ "description": "Use this report to understand query runs across your workspace.",
+ "dataTypesDependencies": ["CommonSecurityLog"],
+ "dataConnectorsDependencies": [],
```
Comment (v-maudan) [inline Workbooks/WorkbooksMetadata.json:1097]: Please Update "dataConnectorsDependencies" as  data connector id value in the data connector(s) on which this workbook depends on. like [ "ForcepointCloudSecurityGateway" ]

**Review by v-maudan — CHANGES_REQUESTED** [review]

**Comment by v-maudan** [comment]
@Dlo-Bagari, Can you resolve the conflicts? Thanks.

File: Workbooks/WorkbooksMetadata.json [hunk 8]
```diff
@@ -1113,6 +1113,22 @@
...
+ "dataTypesDependencies": ["CommonSecurityLog"],
+ "dataConnectorsDependencies": ["ForcepointCloudSecurityGateway"],
+ "previewImagesFileNames": ["ForcepointCloudSecurityGatewayWhite.PNG","ForcepointCloudSecurityGatewayBlack.PNG"],
+ "version": "1.0",
+ "title": "Forcepoint Cloud Security Gateway Workbook",
+ "templateRelativePath": "ForcepointCloudSecuirtyGatewayworkbook.json",
+ "subtitle": "",
+ "provider": "Forcepoint"
+ }
+=======
```
Comment (v-maudan) [inline Workbooks/WorkbooksMetadata.json:1131]: Please resolve conflicts properly,[Remove duplicate code of line no 1131,1132,1133 and update '**,**' at line no 1130]

**Comment by v-jayakal** [comment]
@Dlo-Bagari, as part of the comment resolution, in recent commit, few Notebook files are added in the PR and they are resulting in validation issues. Can you please resolve the issues? you have the following options for resolving:
1. Drop notebook related changes, sync your branch with master to have latest and without conflicts.
2. Create a new branch from master and apply only changes related to connector.
Please let us know if you need any info and resolve the conflicts and validations, Thanks.

**Comment by v-jayakal** [comment]
> @Dlo-Bagari, as part of the comment resolution, in recent commit, few Notebook files are added in the PR and they are resulting in validation issues. Can you please resolve the issues? you have the following options for resolving:
> 
> 1. Drop notebook related changes, sync your branch with master to have latest and without conflicts.
> 2. Create a new branch from master and apply only changes related to connector.
>    Please let us know if you need any info and resolve the conflicts and validations, Thanks.

@Dlo-Bagari - Can you please update the PR? Thanks.

**Comment by Dlo-Bagari** [comment]
> > @Dlo-Bagari, as part of the comment resolution, in recent commit, few Notebook files are added in the PR and they are resulting in validation issues. Can you please resolve the issues? you have the following options for resolving:
> > 
> > 1. Drop notebook related changes, sync your branch with master to have latest and without conflicts.
> > 2. Create a new branch from master and apply only changes related to connector.
> >    Please let us know if you need any info and resolve the conflicts and validations, Thanks.
> 
> @Dlo-Bagari - Can you please update the PR? Thanks.

@v-jayakal, for Forcepoint Cloud Security Gateway onboarding there was nothing to be added to notebooks.
I have created a new branch with the name: origin/Forcepoint-cloud-security-gateway which contains the latest code.
please let me know if that would solve the conflicts.

**Comment by v-maudan** [comment]
@Dlo-Bagari , I am unable to see data connector related changes in new branch which you mentioned- origin/Forcepoint-cloud-security-gateway [PFB screenshot], Could you please create new pull request.
image

**Comment by Dlo-Bagari** [comment]
> @Dlo-Bagari , I am unable to see data connector related changes in new branch which you mentioned- origin/Forcepoint-cloud-security-gateway [PFB screenshot], Could you please create new pull request.
> image

I have created a new PR: onboarding Forcepoint cloud security gateway #1604
do I need to close the previous PR which is:  onboard Forcepoint Cloud Security Gateway #1436

**Comment by v-maudan** [comment]
@Dlo-Bagari , Could you please revert notebook related changes.

**Comment by Dlo-Bagari** [comment]
> @Dlo-Bagari , Could you please revert notebook related changes.

@v-maudan, for Forcepoint Cloud Security Gateway there were no changes made in notebooks, I am not sure what do I need to revert exactly here. can you give me more explanation about what needs to be done?

**Comment by v-jayakal** [comment]
@Dlo-Bagari, the 2 PRs raised for Forcepoint has Notebook files included even though there are no changes to those files. Adding them in PR will corrupt the repo. To explain you the issue, a new PR draft has been created with only the changes of Forcepoint, Please check the PR https://github.com/Azure/Azure-Sentinel/pull/1608, when compared, this new PR does not have notebook files. Can you please create a new PR similar to this PR with only changes related to Forcepoint and without Notebook files. 
Let us know the PR once you have it, we can review and approve it, Thanks.
Forcepoint-Noteboos

**Pull request closed without merging**